docs/USAGE-GUIDE.md

```diff
@@ -77,35 +77,68 @@ And when invoking CMake for your project, do one of two things:
+## Building s2n-tls with Openssl
 
+We keep the build artifacts in the *-build directory:
 ```shell
-# We keep the build artifacts in the -build directory
 cd libcrypto-build
+```
+
+### Download the desired Openssl version:
+Openssl 3.0.5
+```shell
+curl -L -o openssl.tar.gz https://github.com/openssl/openssl/archive/refs/tags/openssl-3.0.5.tar.gz
+tar -xzvf openssl-3.0.5.tar.gz
+cd `tar ztf openssl-3.0.5.tar.gz | head -n1 | cut -f1 -d/`
+```
+
+OpenSSL-1.1.1
+```shell
+curl -LO https://github.com/openssl/openssl/archive/refs/tags/OpenSSL_1_1_1.tar.gz
+tar -xzvf OpenSSL_1_1_1.tar.gz
+cd `tar ztf OpenSSL_1_1_1.tar.gz | head -n1 | cut -f1 -d/`
+```
+
+OpenSSL-1.0.2
+```shell
+curl -LO https://github.com/openssl/openssl/archive/refs/tags/OpenSSL_1_0_2.tar.gz
+tar -xzvf OpenSSL_1_0_2.tar.gz
+cd `tar ztf OpenSSL_1_0_2.tar.gz | head -n1 | cut -f1 -d/`
+```
 
-# Download the latest version of OpenSSL
-curl -LO https://www.openssl.org/source/openssl-1.1.1-latest.tar.gz
-tar -xzvf openssl-1.1.1-latest.tar.gz
+### Build Openssl
+The following config command disables numerous Openssl features and algorithms which are not used
+by s2n-tls. A minimal feature-set can help prevent exposure to security vulnerabilities.
 
-# Build openssl libcrypto
-cd `tar ztf openssl-1.1.1-latest.tar.gz | head -n1 | cut -f1 -d/`
+OpenSSL-1.1.1 and OpenSSL-3.0.5
+```shell
 ./config -fPIC no-shared              \
-         no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-zlib     \
-         no-hw no-mdc2 no-seed no-idea enable-ec_nistp_64_gcc_128 no-camellia\
-         no-bf no-ripemd no-dsa no-ssl2 no-ssl3 no-capieng                  \
-         -DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS      \
-         --prefix=`pwd`/../../libcrypto-root/
+        no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-zlib     \
+        no-hw no-mdc2 no-seed no-idea enable-ec_nistp_64_gcc_128 no-camellia\
+        no-bf no-ripemd no-dsa no-ssl2 no-ssl3 no-capieng                  \
+        -DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS      \
+        --prefix=`pwd`/../../libcrypto-root/
+
 make
 make install
+```
 
-# Build s2n-tls
-cd ../../
+OpenSSL-1.0.2. Mac Users should replace "./config" with "./Configure darwin64-x86_64-cc".
+```shell
+./config -fPIC no-shared              \
+        no-libunbound no-gmp no-jpake no-krb5 no-store    \
+        no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-zlib     \
+        no-hw no-mdc2 no-seed no-idea enable-ec_nistp_64_gcc_128 no-camellia\
+        no-bf no-ripemd no-dsa no-ssl2 no-ssl3 no-capieng                  \
+        -DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS      \
+        --prefix=`pwd`/../../libcrypto-root/
+
+make depend
 make
+make install
 ```
-# Note for 32-bit builds.
-The previous instructions work fine with only a few tweaks to your config command. Example:
+
+OpenSSL-1.1.1 32-bit
 ```shell
 setarch i386 ./config -fPIC no-shared     \
```

Comment:
We should probably put this before 1.0.2, and specify that the first build instructions are for 64-bit.

Reply:
I was doing this in terms of preference order and 32 bit seems like a less likely usecase than 1.0.2 64bit.

64 bit is just the default platform, which I could mention at the very top since it applies to all other instructions also but not sure if its necessary.